query-graphs-js/src/graphPath.ts

```diff
@@ -518,9 +554,57 @@ export class GraphPath<TTrigger, RV extends Vertex = Vertex, TNullEdge extends n
       // is because we only try to re-enter subgraphs for @defer on concrete fields, and so as long as we add downcasts,
       // we should remember that we still need to try re-entering the subgraph.
       deferOnTail: defer ?? (edge && edge.transition.kind === 'DownCast' ? this.props.deferOnTail : undefined),
+      contextToSelection,
+      parameterToContext,
     });
   }
-
+  
+  /**
+   * We are going to grow the conditions by one element with the pathTree on the resolution. Additionally, we may need to merge or replace
+   * the existing elements with elements from the ContextMap
+   */
+  private mergeEdgeConditionsWithResolution(conditionsResolution: ConditionResolution): {
+    edgeConditions: (OpPathTree | null)[],
+    contextToSelection: (Map<string, SelectionSet> | null)[],
+    parameterToContext: (Map<string, ContextAtUsageEntry> | null)[],
+  }{
+    const edgeConditions = this.props.edgeConditions.concat(conditionsResolution.pathTree ?? null);
+    const contextToSelection = this.props.contextToSelection.concat(null);
+    const parameterToContext = this.props.parameterToContext.concat(null);
+    
+    if (conditionsResolution.contextMap === undefined || conditionsResolution.contextMap.size === 0) {
+      return {
+        edgeConditions,
+        contextToSelection,
+        parameterToContext,
+      };
+    }
+    
+    parameterToContext[parameterToContext.length-1] = new Map();
+    
+    for (const [_, entry] of conditionsResolution.contextMap) {
+      const idx = edgeConditions.length - entry.level -1;
+      assert(idx >= 0, 'calculated condition index must be positive');
+      if (edgeConditions[idx] === null) {
+        edgeConditions[idx] = entry.pathTree ?? null;
+      } else if (entry.pathTree !== null) {
+        // here we need to merge the two OpPathTrees. TODO: Do this all at once
+        edgeConditions[idx] = entry.pathTree?.merge(edgeConditions[idx]!) ?? null;
+      }
```

Comment:
This can be simplified to
```typescript
if (entry.pathTree) {
  edgeConditions[idx] = edgeConditions[idx]?.merge(entry.pathTree) ?? entry.pathTree;
}
```